src/Inventory/Request.php

```diff
@@ -349,18 +364,126 @@ public function inventory($data)
     }
 
    /**
-    * Handle agent inventory task request
+    * Handle agent enabled inventory task support on contact request
+    *
+    * @return array
+    */
+    public function handleInventoryTask(): array
+    {
+
+        $params['options']['response'] = [];
+        $params['item'] = $this->inventory->getAgent();
+        $params = Plugin::doHookFunction(Hooks::HANDLE_INVENTORY_TASK, $params);
+
+       // Return inventory task support merging current server/version infos
+        return array_merge($params['options']['response'][self::INVENT_TASK] ?? [], [
```

Comment:
Instead an `array_merge`, i think it should bettter to check only the presence and erase `params` only if answer is invalid.

Plugins hooking in `Plugin::doHookFunction` method receive `$params` in parameter and should return it modified or not.
The hook is usally chained to all plugins declaring it (check https://github.com/g-bougard/glpi/blob/1308dae5131af85bc6bb9360879137d40a707df4/src/Plugin.php#L1473-L1480).

```suggestion
        return isset($params['options']['response'][self::INVENT_TASK]) ? $params : [];
```

Reply:
In your suggestion, you're just missing I wanted to only return `$params['options']['response'][self::INVENT_TASK]` to `handleTask()`, but following your previous suggestion, I'll try to setup `$params` in `handleTask()` and preset `server` and `version` in `$params['options']['response'][self::INVENT_TASK]` so I can just use the `isset` as you're suggesting.
As this hook is actually required for **databaseinventory** plugin, I just need to be sure this won't break anything in that plugin.

Reply:
Resolved in 441904c